Reqnroll/Bindings/BindingMatch.cs

```diff
@@ -14,13 +14,15 @@ public class BindingMatch
         public bool IsScoped { get { return ScopeMatches > 0; } }
 
         public object[] Arguments  { get; private set; }
+        public int[] ArgumentStartOffsets { get; private set; }
         public StepContext StepContext { get; private set; }
 
-        public BindingMatch(IStepDefinitionBinding stepBinding, int scopeMatches, object[] arguments, StepContext stepContext)
+        public BindingMatch(IStepDefinitionBinding stepBinding, int scopeMatches, object[] arguments, int[] argumentStartOffsets, StepContext stepContext)
```

Comment:
Change the `object[] arguments` to `MatchArgument[] arguments` instead.

Reply:
Completed.

Reply:
I think this is not done yet (or a commit was missing).
Currently it is:
```
public BindingMatch(IStepDefinitionBinding stepBinding, int scopeMatches, object[] arguments, int?[] argumentStartOffsets, StepContext stepContext)
```
But it suppose to be:
```
public BindingMatch(IStepDefinitionBinding stepBinding, int scopeMatches, MatchArgument[] arguments, StepContext stepContext)
```
So basically we should merge the `arguments` and the `argumentStartOffsets` parameter to avoid switching based on the type in the `StepDefinitionMatchService` class.

Reply:
OK. See 16723fb.  

Should the MatchArguments be exposed as a property of the BindingMatch? At present, I deconstruct them into the object[] for values and int?[] for StartOffsets that the consumers of BindingMatch currently consume. But modifying them within Reqnroll is not that intrusive - but not sure whether plugins would break?

Reply:
MatchArguments directly exposed as a property of BindingMatch. The deconstructed properties are not. See: [237ec23](https://github.com/reqnroll/Reqnroll/pull/233/commits/237ec23cc4d912cd9738979697e9c845db58e8f8)

Reply:
Super, thx